src/iexamine.cpp

```diff
@@ -2333,6 +2333,139 @@ void iexamine::kiln_full( player &, const tripoint &examp )
     add_msg( _( "It has finished burning, yielding %d charcoal." ), result.charges );
 }
 
+//arc furnance start
+void iexamine::arcfurnace_empty( player &p, const tripoint &examp )
+{
+    furn_id cur_arcfurnace_type = g->m.furn( examp );
+    furn_id next_arcfurnace_type = f_null;
+    if( cur_arcfurnace_type == f_arcfurnace_empty ) {
+        next_arcfurnace_type = f_arcfurnace_full;
+    } else if( cur_arcfurnace_type == f_arcfurnace_metal_empty ) {
+        next_arcfurnace_type = f_arcfurnace_metal_full;
+    } else {
+        debugmsg( "Examined furniture has action arcfurnace_empty, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+
+    static const std::set<material_id> arcfurnaceable{ material_id( "cac2powder" ) };
+    bool fuel_present = false;
+    auto items = g->m.i_at( examp );
+    for( const item &i : items ) {
+        if( i.typeId() == "cac2" ) {
+            add_msg( _( "This furnace already contains calcium carbide." ) );
+            add_msg( _( "Remove it before activating the arc furnace again." ) );
+            return;
+        } else if( i.made_of_any( arcfurnaceable ) ) {
+            fuel_present = true;
+        } else {
+            add_msg( m_bad, _( "This furnace contains %s, which can't be made into calcium carbide!" ), i.tname( 1,
+                     false ) );
+            return;
+        }
+    }
+
+    if( !fuel_present ) {
+        add_msg( _( "This furance is empty. Fill it with powdered coke and lime mix, and try again." ) );
+        return;
+    }
+
+    ///\EFFECT_FABRICATION decreases loss when firing a furnace
+    const int skill = p.get_skill_level( skill_fabrication );
+    int loss = 60 - 2 *
+               skill; // Inefficency is still fine, coal and limestone is abundant
+
+    // Burn stuff that should get charred, leave out the rest
+    units::volume total_volume = 0_ml;
+    for( const item &i : items ) {
+        total_volume += i.volume();
+    }
+
+    auto char_type = item::find_type( "unfinished_cac2" );
+    int char_charges = char_type->charges_per_volume( ( 100 - loss ) * total_volume / 100 );
+    if( char_charges < 1 ) {
+        add_msg( _( "The batch in this furance is too small to yield usable calcium carbide." ) );
+        return;
+    }
+//arc furnaces require a huge amount of current, so 1 full storage battery would work as a stand in
+    if( !p.has_charges( "battery", 40000 ) ) {
```

Comment:
Maybe use >= (not sure what the largest battery is, but there may be larger one's in the future)
Wold this also preclude other power sources (plutonium etc) - I don't know enough about power or power sources to really be commenting here.